primitives/nomad/nomad-base/Cargo.toml

```diff
@@ -0,0 +1,55 @@
+[package]
+name = "nomad-base"
+version = "0.1.0"
+edition = "2021"
+
+# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
+
+[dependencies]
+# Substrate
+codec = { package = "parity-scale-codec", version = "2.0.0", default-features = false, features = ["derive"] }
+scale-info = { version = "1.0", default-features = false, features = ["derive"] }
+sp-std = { version = "4.0.0-dev", default-features = false }
+sp-core = { version = "4.0.0-dev", default-features = false }
+sp-io = { version = "4.0.0-dev", default-features = false }
+frame-support = { version = "4.0.0-dev", default-features = false }
+parity-util-mem = { version = "0.10.2", default-features = false, features = ["primitive-types"] }
+
+primitive-types = { version = "0.10.1", default-features = false, features = ["scale-info", "codec"] }
+tiny-keccak = { version = "2.0.2", default-features = false, features = ["keccak"] }
+signature = { path = "../signature", default-features = false }
+
+nomad-core = { path = "../nomad-core", default-features = false }
+
+[dependencies.serde]
+version = "1.0"
+default-features = false
+optional = true
+features = ["derive"]
+
+[dependencies.ethers-signers]
+version = "0.13.0"
+default-features = false
+optional = true
+
+[dependencies.once_cell]
+version = "1.8.0"
+default-features = false
+optional = true
+
+[features]
+default = ["std"]
+std = [
+	"serde",
+	"primitive-types/serde",
+	"codec/std",
+	"scale-info/std",
+	"frame-support/std",
+	"nomad-core/std"
+]
+testing = [
+	"ethers-signers",
+	"once_cell",
+	"nomad-core/testing",
+	"signature/testing",
+]
```

Comment:
Want to note that the `testing` flag enables some testing-only libraries that must import `std`. It is not possible to include these as `dev-dependencies` because they are still compiled by other crates that import the using crate regardless of `#[cfg(test)]` and break build. In other words, `dev-dependencies` don't bubble up compilation rules to parent crate.

Specifically, `ethers-rs` signing methods are used for testing purposes. `ethers-rs` is not at all `no_std` compatible though so we only compile `ethers-core` if `testing` flag is specified.

Reply:
We should figure out a CI setup that properly sets these flags for the necessary Nomad-related crates. This would include all Nomad primitive crates _except_ `merkle` and all Nomad pallets (in followup PR)

Reply:
@aterentic-ethernal @MiguelDD1, how can we work this into your CI workflow file?

Reply:
@luketchang , wouldn't be enough just to update `.github/workflows/default.yml` in your PR?

```
         uses: actions-rs/cargo@v1
         with:
           command: test
-          args: --workspace --exclude fuzzing
+          args: --workspace --exclude fuzzing --features "testing"
```